graphql_compiler/compiler/emit_sql.py

```diff
@@ -126,6 +134,94 @@ def _find_folded_fields(ir):
+    """
+    if isinstance(element.element, XMLPathBinaryExpression):
+        # this indicates that no label should be inserted for this element
+        kw.update(
+            within_columns_clause=False  # label always gets added if within_columns_clause is True
+        )
+    return compiler.visit_label(element, **kw)
+
+
+def get_xml_path_clause(output_column, where):
+    """Produce an MSSQL-style XML PATH-based aggregation subquery.
+
+    XML PATH clause aggregates the values of the output_column from the output vertex
+    properly encoded to ensure that we can reconstruct a list of the original values
+    regardless of what characters they contain, whether they are empty or null, and
+    regardless of their data type.
+
+    Empty arrays are represented as '', null values are represented as '~', array
+    entries are delimited using '|'.
+
+    All occurrences of '^', '~', and '|' in the original string values are
+    replaced with '^e' (escape), '^n' (null), and '^d' (delimiter), resp.
+
+    Undoing the encoding above as well as the XML reference entity encoding performed
+    by the XML PATH statement is deferred to post-processing when the list is retrieved
+    from the string representation produced by the subquery.
+
+    Post-processing must split on '|', convert '~' to None, and undo both the encoding above
+    and the XML reference entity encoding auto performed by XML PATH. In particular, undo
+    '^d' -> '|', '^e' -> '^', '^n' -> '~', '&amp;' -> '&', '&gt;' -> '>', '&lt;' -> '<',
+    '&0xHEX;' -> u'0xHEX'. XML post processing can be done with
+    https://docs.python.org/3/library/html.html#html.unescape
+
+    Joining from the preceding vertex to the correct output vertex
+    is performed with the WHERE clause. This is contrasted with PostgreSQL subqueries,
+    in which the join from penultimate vertex to output vertex is done within
+    the FROM/JOIN clause of the aggregation subquery.
+
+    Args:
+        output_column: SQLAlchemy Column, value being aggregated with XML PATH
+        where: SQLAlchemy BinaryExpression, predicate used to filter
+
+    Returns:
+        A SQLAlchemy Selectable corresponding to the XML PATH subquery
+    """
+    encoded_column = func.REPLACE(  # replace all occurrences of '^' in the original with '^e'
+        output_column, expression.literal_column('\'^\''), expression.literal_column('\'^e\'')
+    )
+    encoded_column = func.REPLACE(  # replace all occurrences of '~' in the original with '^n'
+        encoded_column, expression.literal_column('\'~\''), expression.literal_column('\'^n\'')
+    )
+    encoded_column = func.REPLACE(  # replace all occurrences of '|' in the original with '^d'
+        encoded_column, expression.literal_column('\'|\''), expression.literal_column('\'^d\'')
+    )
+
+    # delimit elements in the array using '|'and mark nulls with `~`
+    xml_column = (expression.literal_column('\'|\'') + func.COALESCE(  # denote null values with '~'
```

Comment:
Would assigning `expression.literal_column('\'|\'')` to a local (and potentially the same for other literals like it) be allowed? It might make the code easier to read.